### PR TITLE
Refactor align converter

### PR DESCRIPTION
**Pull request by noborus**
Change reflect.DeepEqual to slices.Equal.
rightAlign is changed to not reset every time.
Change trimLeft and trimRight to trimmedIndices.
markNext and markPrev are renamed to nextMark and prevMark.

### CODE REVIEW

File: oviewer/oviewer.go [hunk 1]
```diff
@@ -360,10 +360,15 @@ const QuitSmallCountDown = 10
...
-	esConv    = "es"
-	rawConv   = "raw"
-	alignConv = "align"
+	esConv    string = "es"    // esConv processes escape sequence(default).
+	rawConv   string = "raw"   // rawConv is displayed without processing escape sequences as they are.
+	alignConv string = "align" // alignConv is aligned in each column.
+)
+
+const (
+	generalName string = "general"
```
Comment (ccoVeille) [inline oviewer/oviewer.go:371]: Usually constants are sharing a common prefix, allowing to find the easily when using completion in the IDE, they are also ordered in the godoc if they are exported

```suggestion
	convEscaped    string = "es"    // convEscaped processes escape sequence(default).
	convRaw   string = "raw"   // convRaw is displayed without processing escape sequences as they are.
	convAlign string = "align" // convAlign is aligned in each column.
)

const (
	nameGeneral string = "general"
```

File: oviewer/prepare_draw.go [hunk 2]
```diff
@@ -120,18 +120,19 @@ func (root *Root) setAlignConverter() {
 		maxWidths, addRight = m.maxColumnWidths(maxWidths, addRight, ln)
 	}
 
-	if !reflect.DeepEqual(m.alignConv.maxWidths, maxWidths) {
+	if !slices.Equal(m.alignConv.maxWidths, maxWidths) {
```
Comment (ccoVeille) [inline oviewer/prepare_draw.go:123]: Based of the fact the if starts here and ends with the function, you could do this


```suggestion
	if slices.Equal(m.alignConv.maxWidths, maxWidths) {
		return
	}
```

File: oviewer/action.go [hunk 3]
```diff
@@ -244,8 +244,8 @@ func (root *Root) goLineNumber(lN int) {
 	root.setMessagef("Moved to line %d", lN+1)
 }
 
-// markNext moves to the next mark.
-func (root *Root) markNext(context.Context) {
+// nextMark moves to the next mark.
+func (root *Root) nextMark(context.Context) {
```
Comment (ccoVeille) [inline oviewer/action.go:248]: I wouldn't rename them in the same logic as

https://github.com/noborus/ov/pull/623#discussion_r1749265333

But why not

File: oviewer/oviewer.go (hunk 1 above)
Comment (noborus) [reply oviewer/oviewer.go:371]: Thank you for your comment.
That's right. Change.

File: oviewer/prepare_draw.go (hunk 2 above)
Comment (noborus) [reply oviewer/prepare_draw.go:123]: Oops, that's right.

File: oviewer/action.go (hunk 3 above)
Comment (noborus) [reply oviewer/action.go:248]: This depends on the naming rules of the function name.

Comment (ccoVeille) [reply oviewer/action.go:248]: as I said, why not :grin:

**Pull request merged**